tensorflow/contrib/metrics/python/ops/confusion_matrix_ops.py

```diff
@@ -33,26 +33,27 @@ def confusion_matrix(predictions, labels, num_classes=None, dtype=dtypes.int32,
   Calculate the Confusion Matrix for a pair of prediction and
   label 1-D int arrays.
 
-  Considering a prediction array such as: `[1, 2, 3]`
-  And a label array such as: `[2, 2, 3]`
+  The matrix rows represent the prediction labels and the columns
+  represents the real labels. The confusion matrix is always a 2-D array
+  of shape `[n, n]`, where `n` is the number of valid labels for a given
+  classification task. Both prediction and labels must be 1-D arrays of
+  the same shape in order for this function to work.
 
-  The confusion matrix returned would be the following one:
+  Class labels are expected to start at 0. E.g., if there were
+  three classes then the possible labels would be `[0, 1, 2]`.
 
-  ```python
-      [[0, 0, 0]
-       [0, 1, 0]
-       [0, 1, 0]
-       [0, 0, 1]]
-  ```
+  If `weights` is not `None`, then each prediction contributes its
+  corresponding weight to the total value of the confusion matrix cell.
 
-  If `weights` is not None, then the confusion matrix elements are the
-  corresponding `weights` elements.
+  For example:
 
-  Where the matrix rows represent the prediction labels and the columns
-  represents the real labels. The confusion matrix is always a 2-D array
-  of shape [n, n], where n is the number of valid labels for a given
-  classification task. Both prediction and labels must be 1-D arrays of
-  the same shape in order for this function to work.
+  ```python
+    tf.contrib.metrics.confusion_matrix([1, 2, 3], [2, 2, 3]) ==>
+      [[0, 0, 0, 0]
+       [0, 0, 1, 0]
+       [0, 0, 1, 0]
+       [0, 0, 0, 1]]
+  ```
```

Comment:
Can you add a summary line here saying that since the highest value label was 3, the label space assumed is 0, 1, 2, 3 and hence we end up with n=4 and a 4X4 matrix. I feel that'll make this clearer.